src/graph/validator/GoValidator.cpp

```diff
@@ -69,6 +69,23 @@ Status GoValidator::validateWhere(WhereClause* where) {
   }
 
   auto expr = where->filter();
+  auto filterExprs = ExpressionUtils::collectAll(
+      expr, {Expression::Kind::kVertex, Expression::Kind::kDstProperty});
+  for (auto filterExpr : filterExprs) {
+    if (filterExpr->kind() == Expression::Kind::kDstProperty) {
+      goCtx_->joinDst = true;
+    } else {
+      auto& name = static_cast<const VertexExpression*>(filterExpr)->name();
+      if (name == "VERTEX") {
```

Comment:
Just return false if you dont care this scenario.

Reply:
This is code logic you should check elsewhere than here, IMHO.

Reply:
should report error if user input
`go from ... yield vertex as v`

Reply:
Didn't the previous implementation check that, I mean you shouldn't mess with the logic of checking if there is an endpoint.

Reply:
I think checking whether there is information about the destination vertex and whether to joinDst is logically consistent and does not need to be implemented separately

Reply:
got what's u mean